src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -306,100 +390,185 @@ def _extract_antismash_zip(antismash_obj, project_file_cache):
     if not os.path.exists(output_path):
         os.makedirs(output_path, exist_ok=True)
 
-    antismash_zip = zipfile.ZipFile(antismash_obj.filename)
-    kc_prefix1 = f'{antismash_obj.resolved_id}/knownclusterblast'
-    kc_prefix2 = 'knownclusterblast'
-    for zip_member in antismash_zip.namelist():
-        # TODO other files here?
-        if zip_member.endswith('.gbk') or zip_member.endswith('.json'):
-            antismash_zip.extract(zip_member, path=output_path)
-        elif zip_member.startswith(kc_prefix1) or zip_member.startswith(
-                kc_prefix2):
-            if zip_member.endswith('.txt') and 'mibig_hits' not in zip_member:
+    with zipfile.ZipFile(antismash_obj.filename) as antismash_zip:
+        kc_prefix1 = f'{antismash_obj.resolved_refseq_id}/knownclusterblast'
+        kc_prefix2 = 'knownclusterblast'
+        for zip_member in antismash_zip.namelist():
+            # TODO other files here?
+            if zip_member.endswith('.gbk') or zip_member.endswith('.json'):
                 antismash_zip.extract(zip_member, path=output_path)
+            elif zip_member.startswith(kc_prefix1) or zip_member.startswith(
+                    kc_prefix2):
+                if zip_member.endswith(
+                        '.txt') and 'mibig_hits' not in zip_member:
+                    antismash_zip.extract(zip_member, path=output_path)
 
-    open(os.path.join(output_path, 'completed'), 'w').close()
+    with open(os.path.join(output_path, 'completed'), 'w'):
+        pass
 
     return True
 
 
+def podp_download_and_extract_antismash_data(
+        genome_records: list[Dict[str, Dict[str, str] | str]],
+        project_download_cache: str | PathLike,
+        project_file_cache: str | PathLike):
+
+    genome_status_file = os.path.join(project_download_cache,
```

Comment:
Please use `Pathlib.Path` for all paths in this class and replace `os.path.*` with `Path` methods.